src/portfft/common/bluestein.hpp

```diff
@@ -0,0 +1,76 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's portFFT
+ *
+ **************************************************************************/
+
+#ifndef PORTFFT_COMMON_BLUESTEIN_HPP
+#define PORTFFT_COMMON_BLUESTEIN_HPP
+
+#include "portfft/common/host_dft.hpp"
+#include "portfft/defines.hpp"
+
+#include <complex>
+#include <sycl/sycl.hpp>
+
+namespace portfft {
+namespace detail {
+/**
+ * Utility function to get the dft transform of the chirp signal
+ * @tparam T Scalar Type
+ * @param ptr Host Pointer containing the load/store modifiers.
+ * @param committed_size original problem size
+ * @param dimension_size padded size
+ */
+template <typename T>
+void populate_fft_chirp_signal(T* ptr, std::size_t committed_size, std::size_t dimension_size) {
+  using complex_t = std::complex<T>;
+  std::vector<complex_t> chirp_signal(dimension_size, 0);
+  std::vector<complex_t> chirp_fft(dimension_size, 0);
+  for (std::size_t i = 0; i < committed_size; i++) {
+    double theta = M_PI * static_cast<double>(i * i) / static_cast<double>(committed_size);
+    chirp_signal[i] = complex_t(static_cast<T>(std::cos(theta)), static_cast<T>(std::sin(theta)));
+  }
+  std::size_t num_zeros = dimension_size - 2 * committed_size + 1;
+  for (std::size_t i = 1; i < committed_size; i++) {
+    chirp_signal[committed_size + num_zeros + i - 1] = chirp_signal[committed_size - i];
+  }
+  host_cooley_tukey(chirp_signal.data(), chirp_fft.data(), dimension_size);
```

Comment:
We will be going ahead with host side FFT, because a whole new kernel would need to be JITTED and only for a batch size of 1, unless the size is too large. For a large size of 131072, over 100 runs, my host implementation clocks in at 22.187 ms, which I do not suppose will be faster than JIT + kernel launch + kernel run

For small sizes like 256: this host_cooley_tukey implementation clocks in at 23.4415 microseconds

Reply:
That is a good argument, I am just concerned about additional code that we will need to maintain.

Reply:
I have written in generically enough and tested it rigorously that I suppose it should be alright.

Reply:
What I am essentially trying to say is that we can merge and forget about it. I do not see a scenario where we would need to actively maintain it

Reply:
You can not know ahead of time what will need to be maintained. For example we might decide in the future that this needs to be optimized. Or we find a bug in it.

I still think this is a bad idea, but resolving, as I do not have any better option to deal with this.